Create mainspower.yaml

Darthmonkey opened the pull request:
mainspower.yaml is an example of a 3 phase yaml file for use with the 'Beginner Instructions'.

timmills commented:
Hi, this has been really helpful in getting one of these devices up and running in HomeAssistant. Ive added a bit extra to make it show up in the Energy section of HomeAssistant. Let me know if you want to review the changes i've made/added and perhaps add it to your code (im not confident in my testing just yet)

nlscarter commented:
This was great, I edited it a little to give me 4 seperate kWh readings in homeassistant on single phase

timmills commented:
The code had 3 phases combined for total incoming power I;m then using the
4th to monitor hot water power being used. Did you also enable the energy
component so it shows up in the Energy section?

On Wed, Nov 2, 2022 at 11:49 PM nlscarter ***@***.***> wrote:

> This was great, I edited it a little to give me 4 seperate kWh readings in
> homeassistant on single phase
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Mottramlabs/4-Channel-Mains-Current-Sensor-ESP8266/pull/7#issuecomment-1300340070>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAHWNTH7WZCSKBB5TFZZQPTWGJPPLANCNFSM6AAAAAAQ3J2WFI>
> .
> You are receiving this because you commented.Message ID:
> <Mottramlabs/4-Channel-Mains-Current-Sensor-ESP8266/pull/7/c1300340070@
> github.com>
>

Darthmonkey commented:
@timmills I haven't altered the code to enable use in the 'Energy' section, it's been showing up in there for me since I added it via  Settings | Dashboards | Energy | Electricity Grid | Add Consumption | 'mainspower Total kWh'. I have a HACS plugin which queries my Fronius Solar Inverter, and I added that via .... | Add Return | 'Fronius Day Energy'.
In addition, I'm not sure what changes you've made to mainspower.yaml, I've compared the one in 'Files Changed' to my existing one, and can't see any changes.

timmills commented:
Hi, Im setting up another unit with 30A CT sensors, do you know where you
could modify the code to use these or do you just recalibrate - I'm getting
some pretty high readings with zero load.
I had added these types of codes to bring it into the consumption section -
maybe it would have worked with out it?
Energy Sensors (kWh) for HA Dashboard
  - platform: total_daily_energy
    name: ${esp_id} Total kWh Unit 1
    power_id: Unit1Watts
    filters:
      - multiply: 0.001
    unit_of_measurement: kWh

On Mon, Nov 7, 2022 at 3:31 PM Darthmonkey ***@***.***> wrote:

> @timmills <https://github.com/timmills> I haven't altered the code to
> enable use in the 'Energy' section, it's been showing up in there for me
> since I added it via Settings | Dashboards | Energy | Electricity Grid |
> Add Consumption | 'mainspower Total kWh'. I have a HACS plugin which
> queries my Fronius Solar Inverter, and I added that via .... | Add Return |
> 'Fronius Day Energy'.
> In addition, I'm not sure what changes you've made to mainspower.yaml,
> I've compared the one in 'Files Changed' to my existing one, and can't see
> any changes.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Mottramlabs/4-Channel-Mains-Current-Sensor-ESP8266/pull/7#issuecomment-1305068069>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAHWNTCZOUC4H3JOCSIZR23WHCAYZANCNFSM6AAAAAAQ3J2WFI>
> .
> You are receiving this because you were mentioned.Message ID:
> <Mottramlabs/4-Channel-Mains-Current-Sensor-ESP8266/pull/7/c1305068069@
> github.com>
>

dacorn commented:
Hi guys, I am using the yaml you created but having some trouble. First I am using board version3 just in case this is my problem.
I set it up, calibrated it no problem. Moved the unit to my electrical unit plugged it back in, nothing connected just powered on the board. All 4 clamps are still showing the value of what I was using on my test bench for the calibration. I have restarted the unit, recompiled the YAML and uploaded it. Same the value is shows in the test page and in home assistant show the values from when I calibrated it.  In the trace logs I see the raw values are showing 0 which is correct but the sensor value holds on to the calibration value. Any ideas? Thanks in advance.

timmills commented:
Can you post a copy of the Yaml - you didn’t set the 0 value to the calibrated value rather than the value it was reading to the actual value?Sent from my iPhone.On 17 Dec 2022, at 5:31 pm, dacorn ***@***.***> wrote:﻿
Hi guys, I am using the yaml you created but having some trouble. First I am using board version3 just in case this is my problem.
I set it up, calibrated it no problem. Moved the unit to my electrical unit plugged it back in, nothing connected just powered on the board. All 4 clamps are still showing the value of what I was using on my test bench for the calibration. I have restarted the unit, recompiled the YAML and uploaded it. Same the value is shows in the test page and in home assistant show the values from when I calibrated it.  In the trace logs I see the raw values are showing 0 which is correct but the sensor value holds on to the calibration value. Any ideas? Thanks in advance.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

dacorn commented:
Hi, please see the code below, the calibration is commented out as I was just trying to make changes but this is as it is running on my device. 

Thanks
Dave
[energy_4ch.yaml.txt](https://github.com/Mottramlabs/4-Channel-Mains-Current-Sensor-ESP8266/files/10253029/energy_4ch.yaml.txt)

timmills commented:
There is an uncommented filters without a value on the 2nd sensor - maybe
that could be messing it up?

What if you put the filters back in with 0 -> 0

On Sun, Dec 18, 2022 at 1:01 PM dacorn ***@***.***> wrote:

> Hi, please see the code below, the calibration is commented out as I was
> just trying to make changes but this is as it is running on my device.
>
> Thanks
> Dave
> energy_4ch.yaml.txt
> <https://github.com/Mottramlabs/4-Channel-Mains-Current-Sensor-ESP8266/files/10253029/energy_4ch.yaml.txt>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Mottramlabs/4-Channel-Mains-Current-Sensor-ESP8266/pull/7#issuecomment-1356625712>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAHWNTB6OKK7VXDHKZN54TLWNZV7ZANCNFSM6AAAAAAQ3J2WFI>
> .
> You are receiving this because you were mentioned.Message ID:
> <Mottramlabs/4-Channel-Mains-Current-Sensor-ESP8266/pull/7/c1356625712@
> github.com>
>

Jarled commented:
> Hi, please see the code below, the calibration is commented out as I was just trying to make changes but this is as it is running on my device.
> 
> Thanks Dave [energy_4ch.yaml.txt](https://github.com/Mottramlabs/4-Channel-Mains-Current-Sensor-ESP8266/files/10253029/energy_4ch.yaml.txt)

Is this working now, as i would like to use it or should i use
[Create mainspower.yaml](https://github.com/Mottramlabs/4-Channel-Mains-Current-Sensor-ESP8266/pull/7/commits/f388696031f898e13d495d2b4f584ed8a2fae0e0)

dacorn commented:
Which version do you have? Mine is version 3 on green board. It will run but the load will not go to zero when there is no current no matter if it’s calibrated or not. This is of course with the calibration values in commented. Regards,DaveOn Dec 23, 2022, at 2:11 AM, Jarled ***@***.***> wrote:﻿

Hi, please see the code below, the calibration is commented out as I was just trying to make changes but this is as it is running on my device.
Thanks Dave energy_4ch.yaml.txt

Is this working now, as i would like to use it or should i use
Create mainspower.yaml

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

Jarled commented:
> Which version do you have? Mine is version 3 on green board. It will run but the load will not go to zero when there is no current no matter if it’s calibrated or not. This is of course with the calibration values in commented. Regards,DaveOn Dec 23, 2022, at 2:11 AM, Jarled ***@***.***> wrote:﻿ Hi, please see the code below, the calibration is commented out as I was just trying to make changes but this is as it is running on my device. Thanks Dave energy_4ch.yaml.txt Is this working now, as i would like to use it or should i use Create mainspower.yaml —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

I have the version 3 blue board and i think i have made a mistake and have bought 4 x YHDC scT-013-000v and not the 50ma version. can this still work ?
